messagingpush/src/main/java/io/customer/messagingpush/CustomerIOFirebaseMessagingReceiver.kt

```diff
@@ -0,0 +1,26 @@
+package io.customer.messagingpush
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import io.customer.messagingpush.di.moduleConfig
+import io.customer.messagingpush.di.pushTrackingUtil
+import io.customer.messagingpush.extensions.getSDKInstanceOrNull
+import io.customer.sdk.data.request.MetricEvent
+
+class CustomerIOFirebaseMessagingReceiver : BroadcastReceiver() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+        val extras = intent.extras ?: return
+        // if CustomerIO instance isn't initialized, we cannot process the notification
+        val sdkInstance = context.getSDKInstanceOrNull() ?: return
+
+        val moduleConfig = sdkInstance.diGraph.moduleConfig
+        val pushTrackingUtil = sdkInstance.diGraph.pushTrackingUtil
+
+        // Track delivered event only if auto-tracking is enabled
+        if (moduleConfig.autoTrackPushEvents) {
+            pushTrackingUtil.parseAndTrackMetricEvent(extras, MetricEvent.delivered)
+        }
```

Comment:
Can we add automated tests to this code here? Help us confirm that we will only track metrics if the customer has enabled it.

Reply:
We unfortunately don't currently support `autoTrackPushEvents` in case of auto-initialization. So this may not work well with wrapper SDKs if the app was killed. I have created a [bug](https://github.com/customerio/issues/issues/10254) for this so this doesn't get ignored. Meanwhile, I'm writing tests to see how can we test both normal and auto-initialization, I'll push them soon.

Reply:
Looking at the code changes, I think the [bug linked](https://github.com/customerio/issues/issues/10254) is the right ticket to write tests for testing this with auto-initialization. For now, I have added tests that covers metrics tracking for `PushMessageProcessor`.